Lib/zipfile/_path.py

```diff
@@ -258,7 +258,8 @@ def open(self, mode='r', *args, pwd=None, **kwargs):
                 raise ValueError("encoding args invalid for binary operation")
             return stream
         else:
-            kwargs["encoding"] = io.text_encoding(kwargs.get("encoding"))
+            if "encoding" in kwargs:
+                kwargs["encoding"] = io.text_encoding(kwargs["encoding"])
```

Comment:
this logic isn't quite right, we should accept it as our own argument so io.text_encoding is called upon it even when positional.